### PR TITLE
CDEV-55: Enhance Component Empty States

### PR DESCRIPTION
**Pull request by sarahmangels**
Update the empty display to differentiate between no query results vs no results shown due to filters.

https://zeushealth.atlassian.net/browse/CDEV-55

image
image

### CODE REVIEW

File: src/components/core/empty-table.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export type EmptyTableProps = {
+  resourceName: string;
+  isEmptyQuery: boolean;
+  isEmptyFilters: boolean;
+};
+
+export const EmptyTable = ({ resourceName, isEmptyQuery, isEmptyFilters }: EmptyTableProps) => {
+  if (isEmptyQuery) {
+    return (
+      <div>
```
Comment (WesleyKapow) [inline src/components/core/empty-table.tsx:13]: nit: You can add `ctw-flex ctw-justify-center` to the top div and that should allow you to simplify some of the stuff below.

other nit: In general, adding whitespace between elements, horizontal or vertical, should be done through a parent element using [taildwind's spacing](https://tailwindcss.com/docs/space) classes, e.g. `ctw-space-y-3`. While you could add space above an element, or space below an element to achieve the same thing, semantically it makes more sense to say "these children should be spaced X amount" as appose to "I want X space above me".

File: src/components/core/empty-table.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        </div>
+        <div className="ctw-pt-6 ctw-text-center ctw-text-xl ctw-font-medium">
+          No matching {resourceName} found
+        </div>
+        <div className="ctw-pt-3 ctw-text-center ctw-text-base ctw-font-normal">
+          Try changing the filters.
+        </div>
+      </div>
+    );
+  }
```
Comment (WesleyKapow) [inline src/components/core/empty-table.tsx:40]: nit: This has the exact same structure as above. You could refactor to a helper component.

File: src/components/core/empty-table.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  return <div>blah</div>;
```
Comment (WesleyKapow) [inline src/components/core/empty-table.tsx:41]: uhoh

File: src/components/core/empty-table.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { faInbox, faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+export type EmptyTableProps = {
+  resourceName: string;
+  isEmptyQuery: boolean;
+  isEmptyFilters: boolean;
```
Comment (WesleyKapow) [inline src/components/core/empty-table.tsx:7]: nit: Names here are a bit confusing, I was first thinking `isEmptyFilters` meant there are no filters applied.

How about `hasZeroRecords`, `hasZeroFilteredRecords`?

**Review by WesleyKapow — APPROVED** [review]
Bunch of nits, otherwise LGTM

File: src/components/core/empty-table.tsx (hunk 1 above)
Comment (sarahmangels) [reply src/components/core/empty-table.tsx:13]: @WesleyKapow how do i handle that when the spacing isnt equal. the designs have 24px between the icon and large text, and 12 btwn large text and small text

File: src/components/core/empty-table.tsx (hunk 4 above)
Comment (sarahmangels) [reply src/components/core/empty-table.tsx]: addressed here https://github.com/zeus-health/ctw-component-library/pull/887/commits/435ce8d8daa14b9f1279f0cc4e7267316b694be1

File: src/components/core/empty-table.tsx (hunk 3 above)
Comment (sarahmangels) [reply src/components/core/empty-table.tsx]: fixed https://github.com/zeus-health/ctw-component-library/pull/887/commits/435ce8d8daa14b9f1279f0cc4e7267316b694be1

File: src/components/core/empty-table.tsx (hunk 2 above)
Comment (sarahmangels) [reply src/components/core/empty-table.tsx]: simplified here https://github.com/zeus-health/ctw-component-library/pull/887/commits/435ce8d8daa14b9f1279f0cc4e7267316b694be1

File: src/components/core/empty-table.tsx (hunk 1 above)
Comment (WesleyKapow) [reply src/components/core/empty-table.tsx]: You'd setup two wrappers. One for everything that has 12 and then one between that div and the one with 24

```jsx
<div className="ctw-space-y-6">
  <icon>
  <div className="ctw-space-y-3">
    <div>no matching...</div>
   <div>try changing filters</div>
</div>
```

**Review by WesleyKapow — APPROVED** [review]

**Review by WesleyKapow — DISMISSED** [review]

**Review by pmadria3 — APPROVED** [review]

**Pull request merged**